05_stringer/runnerdave/main_test.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("127.0.0.1\n")
+	actual := strconv.Quote(buf.String())
+	t.Logf("expected:%s", expected)
+	t.Logf("actual:%s", actual)
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()")
+	}
+}
+
+func TestStringer(t *testing.T) {
+	tables := []struct {
+		input  IPAddr
+		output string
+	}{
+		{IPAddr{127, 0, 0, 1}, "127.0.0.1"},
+		{IPAddr{10, 0, 0, 1}, "10.0.0.1"},
+		{IPAddr{191, 255, 255, 255}, "191.255.255.255"},
+		{IPAddr{10, 10}, "10.10.0.0"},
+		{IPAddr{}, "0.0.0.0"},
+	}
+
+	for _, table := range tables {
+		expected := table.output
+		actual := table.input.String()
+		if expected != actual {
+			t.Errorf("Stringer of (%d) was incorrect, got: %s, want: %s.", table.input, actual, expected)
```

Comment:
Sorry I didn't notice this before, but the `%d` format specifier here is incorrect. I would suggest `%#v` which will give a Go-syntax representation of the value (https://golang.org/pkg/fmt/) as shown here: https://goplay.space/#BiRitf92ruM (`%v` would call the `String()` method which is counter-productive here)

Reply:
implemented